data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineModel.java

```diff
@@ -64,6 +65,9 @@ else if (preppers != null) {
     @JsonInclude(JsonInclude.Include.NON_NULL)
     private final PluginModel buffer;
 
+    @JsonProperty("router")
```

Comment:
Should this have the `@JsonInclude(JsonInclude.Include.NON_NULL` annotation? It is an optional configuration similar to `buffer` and `processor`

Reply:
I updated this to use `NON_EMPTY`. This will not serialize an empty list. This makes null and empty list semantically the same and allows Java clients to not have to worry about null.

I create #1669 to make some related changes. In that, I will change `processor` to also have the same behavior. This should make it easier to code against these Java models (no null lists).